components/movementsensor/wheeledodometry/wheeledodometry.go

```diff
@@ -212,8 +215,19 @@ func (o *odometry) Reconfigure(ctx context.Context, deps resource.Dependencies,
 
 	o.orientation.Yaw = 0
 	o.originCoord = geo.NewPoint(0, 0)
+	o.coordUpToDate.Store(false)
+	o.mu.Unlock()
+	defer o.mu.Lock() // Must be unlocked for trackPosition. We put a Lock on the defer stack so the earlier deferred unlock does not hang.
 	o.trackPosition() // (re-)initializes o.workers
-
+	// Wait for trackPosition to initialize coord so we do not start with stale data
+	for !o.coordUpToDate.Load() {
+		select {
+		case <-ctx.Done():
+			return ctx.Err()
+		default:
+		}
+		time.Sleep(time.Duration(o.timeIntervalMSecs) * time.Millisecond)
```

Comment:
[Note to self], we should see if the control package kick back is affected by this change. cc @JohnN193